subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -260,9 +258,8 @@ public class DefaultConfiguration extends AbstractFileCollection implements Conf
 
     private Action<? super ConfigurationInternal> beforeLocking;
 
-    private List<String> declarationAlternatives;
-    private DeprecationMessageBuilder.WithDocumentation consumptionDeprecation;
-    private List<String> resolutionAlternatives;
+    private List<String> declarationAlternatives = ImmutableList.of();
+    private List<String> resolutionAlternatives = ImmutableList.of();
```

Comment:
❔ I know these all use the same `EMPTY` instance.  But since these fields will be rarely used, would it be worth leaving them initialized to `null` and delaying initializing them until usage, just to shave off a few more bytes?

Reply:
I don't think that will make a difference. A reference to some shared field and a null reference should take up the same amount of memory. The `EMPTY` instance is already allocated and is without a doubt used elsewhere.